tidy3d/components/medium.py

```diff
@@ -1546,6 +1553,146 @@ def to_medium(self) -> Medium:
             frequency_range=self.frequency_range,
         )
 
+    @staticmethod
+    def _eV_to_angular_freq(f_eV: float):
+        """Convert frequency in unit of eV to rad/s.
+
+        Parameters
+        ----------
+        f_eV : float
+            Frequency in unit of eV
+        """
+        return f_eV / HBAR
+
+    @staticmethod
+    def _angular_freq_to_eV(f_rad: float):
+        """Convert frequency in unit of rad/s to eV.
+
+        Parameters
+        ----------
+        f_rad : float
+            Frequency in unit of rad/s
+        """
+        return f_rad * HBAR
+
+    @staticmethod
+    def _angular_freq_to_Hz(f_rad: float):
+        """Convert frequency in unit of rad/s to Hz.
+
+        Parameters
+        ----------
+        f_rad : float
+            Frequency in unit of rad/s
+        """
+        return f_rad / 2 / np.pi
+
+    @staticmethod
+    def _Hz_to_angular_freq(f_hz: float):
+        """Convert frequency in unit of Hz to rad/s.
+
+        Parameters
+        ----------
+        f_hz : float
+            Frequency in unit of Hz
+        """
+        return f_hz * 2 * np.pi
+
+    @cached_property
+    def _imag_ep_extrema(self) -> Tuple[ArrayFloat1D]:
```

Comment:
We might want this function to be public. For example, it can be useful to have a certificate (in the form of the extrema frequency) for the min loss and max loss in a model.

Reply:
Could you elaborate more?

Reply:
I see the static method is public now. I mean that, if you are really working hard to get a good material fit, it can be useful to relax the passivity constraint so some small negative min_loss, and see where the resulting material fit violates passivity (not just the min_loss, but also the frequencies where it can occur.) Then, you might want to change something (such as shrinking the frequency range) in order to avoid this issue.

Reply:
i see, that'll be something in the future but not in this PR?